Code/src/main/java/nl/utwente/group10/ui/components/anchors/InputAnchor.java

```diff
@@ -1,42 +1,46 @@
 package nl.utwente.group10.ui.components.anchors;
 
-import java.util.Optional;
-
-import edu.emory.mathcs.backport.java.util.Arrays;
 import nl.utwente.group10.haskell.expr.Expr;
 import nl.utwente.group10.haskell.expr.Ident;
-import nl.utwente.group10.haskell.type.Type;
-import nl.utwente.group10.ui.CustomUIPane;
 import nl.utwente.group10.ui.components.blocks.Block;
-import nl.utwente.group10.ui.components.blocks.InputBlock;
 import nl.utwente.group10.ui.components.lines.Connection;
-import nl.utwente.group10.ui.exceptions.TypeUnavailableException;
 import nl.utwente.group10.ui.handlers.AnchorHandler;
 
 /**
- * Anchor that specifically functions as an input.
+ * ConnectionAnchor that specifically functions as an input.
  */
 public class InputAnchor extends ConnectionAnchor {
+    /** The expression to return when there is no connection. */
+    private Expr connectionlessExpr;
```

Comment:
Case:

1. Not connected. Expr: ```undefined```. Type: ```a```
2. Connect output (example: ```sin```). Type: ```a:Floating```.
3. Disconnect output. The unified thing stays around.